client/components/Deposit/Deposit.tsx

```diff
@@ -0,0 +1,213 @@
+import { Callout } from '@blueprintjs/core';
+import Doi from 'client/containers/DashboardSettings/PubSettings/Doi';
+import { apiFetch } from 'client/utils/apiFetch';
+import {
+	getFirstIntraWorkRelationship,
+	Resource,
+	resourceKindToProperNoun,
+} from 'deposit/resource';
+import React, { useEffect, useState } from 'react';
+import { Collection, DepositTarget, InitialCommunityData, Pub } from 'types';
+import { assert, exists } from 'utils/assert';
+import { PUBPUB_DOI_PREFIX } from 'utils/crossref/communities';
+import DataciteDeposit from './DataciteDeposit';
+import './deposit.scss';
+import { UpdateDoi } from './UpdateDoi';
+
+type PubProps = {
+	pub: Pub;
+	updatePub: (...args: any[]) => any;
+};
+
+type CollectionProps = {
+	collection: Collection;
+	updateCollection: (...args: any[]) => any;
+};
+
+type Props = {
+	canIssueDoi: boolean;
+	communityData: InitialCommunityData;
+	depositTarget?: DepositTarget;
+} & (PubProps | CollectionProps);
+
+function isSupplementTo(resource: Resource) {
+	return resource.relationships.some(
+		(relationship) => relationship.relation === 'Supplement' && relationship.isParent,
+	);
+}
+
+function extractDoiSuffix(doi: string, depositTarget?: DepositTarget) {
+	const prefix = depositTarget?.doiPrefix ?? PUBPUB_DOI_PREFIX;
+	return doi.replace(`${prefix}/`, '');
+}
+
+export default function Deposit(props: Props) {
```

Comment:
This component chooses between the existing Crossref deposit UI and the new Datacite UI. Some of the code was extracted to be shared between both implementations, but more work will need to be done to make them isomorphic.